src/asciidoc/core-beans.adoc

```diff
@@ -4562,14 +4562,16 @@ automatically resolved, with no special setup necessary.
 	}
 ----
 
-[NOTE]
+[WARNING]
 ====
-`@Autowired`, `@Inject`, `@Resource`, and `@Value` annotations are handled by Spring
+`@Autowired`, `@Inject`, `@Resource`, and `@Value`  annotations are handled by Spring
 `BeanPostProcessor` implementations which in turn means that you __cannot__ apply these
 annotations within your own `BeanPostProcessor` or `BeanFactoryPostProcessor` types (if
 any). These types must be 'wired up' explicitly via XML or using a Spring `@Bean` method.
-====
 
+Also, do not use `@Autowired`, `@Inject`, and `@Resource` annotations within a BeanFactory class nor any of its dependencies. Doing so can result in a `BeanCurrentlyInCreationException`. The `@Value` annotation may also cause the problem, d
+epending on whether it causes a bean to be instantiated.
+====
```

Comment:
that looks like a complete different conversation to me. What do you mean by "within a `BeanFactory`". I wouldn't expect anyone to use those annotations there (and that's not very typical in use code anyway).

The `@Value` is yet another thing.